context/app/static/js/pages/Organ/hooks.ts

```diff
@@ -0,0 +1,154 @@
+import { useMemo } from 'react';
+import useSearchData, { useSearchHits } from 'js/hooks/useSearchData';
+import { Dataset } from 'js/components/types';
+import { SearchHit } from '@elastic/elasticsearch/lib/api/types';
+import { mustHaveOrganClause } from './queries';
+
+export function useHasSamplesQuery(searchItems: string[]) {
+  const samplesQuery = useMemo(
+    () => ({
+      query: {
+        bool: {
+          must: [
+            {
+              term: {
+                'entity_type.keyword': 'Sample',
+              },
+            },
+            mustHaveOrganClause(searchItems),
+          ],
+        },
+      },
+      _source: false,
+      size: 1,
+    }),
+    [searchItems],
+  );
+
+  const { searchHits: samplesHits } = useSearchHits(samplesQuery);
+  return samplesHits;
+}
+
+interface Bucket {
+  key: string;
+  doc_count: number;
+}
+
+interface AssayBucketAggregations {
+  mapped_data_types: {
+    'assay_display_name.keyword': {
+      buckets: Bucket[];
+    };
+  };
+}
+
+export function useAssayBucketsQuery(searchItems: string[]) {
+  const assaysQuery = useMemo(
+    () => ({
+      size: 0,
+      aggs: {
+        mapped_data_types: {
+          filter: {
+            bool: {
+              must: [
+                {
+                  term: {
+                    'entity_type.keyword': 'Dataset',
+                  },
+                },
+                mustHaveOrganClause(searchItems),
+              ],
+            },
+          },
+          aggs: {
+            'assay_display_name.keyword': { terms: { field: 'assay_display_name.keyword', size: 1000 } },
```

Comment:
We're unlikely to hit the ceiling of 1000, but should we just max it to 10000 now?